machine/v1beta1/types_azureprovider.go

```diff
@@ -215,6 +223,48 @@ type OSDisk struct {
 	CachingType string `json:"cachingType,omitempty"`
 }
 
+// DataDisk specifies the parameters that are used to add one or more data disks to the machine.
+type DataDisk struct {
+	// NameSuffix is the suffix to be appended to the machine name to generate the disk name.
+	// Each disk name will be in format <machineName>_<nameSuffix>.
+	// NameSuffix name must start and finish with an alphanumeric character and can only contain letters, numbers, underscores, periods or hyphens.
+	// The overall disk name must not exceed 80 chars in length.
+	// +kubebuilder:validation:Pattern:=`^[a-zA-Z0-9](?:[\w\.-]*[a-zA-Z0-9])?$`
+	// +kubebuilder:validation:MaxLength:=78
+	// +kubebuilder:validation:Required
+	NameSuffix string `json:"nameSuffix"`
+	// DiskSizeGB is the size in GB to assign to the data disk.
+	// +kubebuilder:validation:Minimum=4
+	// +kubebuilder:validation:Required
+	DiskSizeGB int32 `json:"diskSizeGB"`
+	// ManagedDisk specifies the Managed Disk parameters for the data disk.
+	// +optional
+	ManagedDisk ManagedDiskParameters `json:"managedDisk,omitempty"`
+	// Lun Specifies the logical unit number of the data disk. This value is used to identify data disks within the VM and therefore must be unique for each data disk attached to a VM.
```

Comment:
Why do we let a user set this instead of choosing it ourselves?

Reply:
This can be set both in the Azure Portal and in the upstream `kubernetes-sigs/cluster-api-provider-azure` implementation. So we thought we might want to favor compatibility with those and preserve the ability to manually specify them.

That said, if these `LUN`s are not set (and I would expect most users to do so), they will be be automatically chosen by the controller.

Reply:
> So we thought we might want to favor compatibility with those and preserve the ability to manually specify them.

Is there a some user capability that requires this.  Exposure in other APIs doesn't mean it should be exposed in ours.

Reply:
> Although this doesn't rule out that there might be certain use cases where being able to specify a LUN it's a hard requirement.

Do we have any?  A "might be useful someday" isn't sufficient to provide an API field that could be added later if truly required.

Reply:
Not really at the moment. Ok if you think we can add it later if truly required, I'm happy to drop it.